src/api/rpc/full_node/index.ts

```diff
@@ -503,6 +503,162 @@ export async function get_fee_estimate<T extends TRPCAgent | TDaemon>(agent: T,
+  return agent.sendMessage<R>(chia_full_node_service, farm_block_command, data);
+}
+
+
+
+export const set_auto_farming_command = "set_auto_farming";
+export type set_auto_farming_command = typeof set_auto_farming_command;
+export type TSetAutoFarmingRequest = {
+  auto_farm: bool;
+};
+export type TSetAutoFarmingResponse = {
+  auto_farm_enabled: bool;
+  success: bool;
+};
+export type WsSetAutoFarmingMessage = GetMessageType<chia_full_node_service, set_auto_farming_command, TSetAutoFarmingResponse>;
+export async function set_auto_farming<T extends TRPCAgent | TDaemon>(agent: T, data: TSetAutoFarmingRequest){
+  type R = ResType<T, TSetAutoFarmingResponse, WsSetAutoFarmingMessage>;
+  return agent.sendMessage<R>(chia_full_node_service, set_auto_farming_command, data);
+}
+
+
+
+export const get_auto_farming_command = "get_auto_farming";
+export type get_auto_farming_command = typeof get_auto_farming_command;
+export type TGetAutoFarmingRequest = {
+};
+export type TGetAutoFarmingResponse = {
+  new_peak_height: uint64;
+  success: bool;
+};
+export type WsGetAutoFarmingMessage = GetMessageType<chia_full_node_service, get_auto_farming_command, TGetAutoFarmingResponse>;
+export async function get_auto_farming<T extends TRPCAgent | TDaemon>(agent: T, data: TGetAutoFarmingRequest){
+  type R = ResType<T, TGetAutoFarmingResponse, WsGetAutoFarmingMessage>;
+  return agent.sendMessage<R>(chia_full_node_service, get_auto_farming_command, data);
+}
+
+
+
+export const get_farming_ph_command = "get_farming_ph";
+export type get_farming_ph_command = typeof get_farming_ph_command;
+export type TGetFarmingPhRequest = {
+};
+export type TGetFarmingPhResponse = {
+  puzzle_hash: str;
+  success: bool;
+};
+export type WsGetFarmingPhMessage = GetMessageType<chia_full_node_service, get_farming_ph_command, TGetFarmingPhResponse>;
+export async function get_farming_ph<T extends TRPCAgent | TDaemon>(agent: T, data: TGetFarmingPhRequest){
+  type R = ResType<T, TGetFarmingPhResponse, WsGetFarmingPhMessage>;
+  return agent.sendMessage<R>(chia_full_node_service, get_farming_ph_command, data);
+}
+
+
+
+export const get_all_coins_command = "get_all_coins";
+export type get_all_coins_command = typeof get_all_coins_command;
+export type TGetAllCoinsRequest = {
+  include_spent_coins?: bool;
+};
+export type TGetAllCoinsResponse = {
+  coin_records: CoinRecord[];
+  success: bool;
+};
+export type WsGetAllCoinsMessage = GetMessageType<chia_full_node_service, get_all_coins_command, TGetAllCoinsResponse>;
+export async function get_all_coins<T extends TRPCAgent | TDaemon>(agent: T, data: TGetAllCoinsRequest){
+  type R = ResType<T, TGetAllCoinsResponse, WsGetAllCoinsMessage>;
+  return agent.sendMessage<R>(chia_full_node_service, get_all_coins_command, data);
+}
+
+
+
+export const get_all_puzzle_hashes_command = "get_all_puzzle_hashes";
+export type get_all_puzzle_hashes_command = typeof get_all_puzzle_hashes_command;
+export type TGetAllPuzzleHashesRequest = {
+};
```

Comment:
In `chia-agent`, definition of empty request is not needed.
It's OK to remove these lines.